src/station_cmd.cpp

```diff
@@ -2200,20 +2200,27 @@ uint8 GetAirportNoiseLevelForTown(const AirportSpec *as, TileIterator &it, TileI
  */
 Town *AirportGetNearestTown(const AirportSpec *as, const TileIterator &it)
 {
+	if (Town::GetNumItems() == 0) return NULL;
+
 	Town *t, *nearest = NULL;
 	uint add = as->size_x + as->size_y - 2; // GetMinimalAirportDistanceToTile can differ from DistanceManhattan by this much
 	uint mindist = UINT_MAX - add; // prevent overflow
-	FOR_ALL_TOWNS(t) {
-		if (DistanceManhattan(t->xy, it) < mindist + add) { // avoid calling GetMinimalAirportDistanceToTile too often
-			TileIterator *copy = it.Clone();
-			uint dist = GetMinimalAirportDistanceToTile(*copy, t->xy);
-			delete copy;
+
+	TileIterator *copy = it.Clone();
+	for (TileIndex cur_tile = *copy; cur_tile != INVALID_TILE; cur_tile = ++*copy) {
+		t = CalcClosestTownFromTile(cur_tile, mindist + add);
+		if (t != NULL) {
+			uint dist = DistanceManhattan(t->xy, cur_tile);
+			if (dist == mindist) {
+				if (t->index < nearest->index) nearest = t;
+			}
```

Comment:
`if (dist == mindist && ...)`

Reply:
Also add a comment about why the lowest index is preferred.

Reply:
I don't know how to comment this.

Reply:
Something like:

> Previous behaviour used a town-scan and thus preferred towns with a lower index.
> To replicate this we also consider near towns with the same distance and compare their index.

Note that noise level is not saved, so it isn't actually required keep this behaviour.